File: java/BlowfishCipher.java
package com.example.crypto;

import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;

/** BlowfishCipher -- synthetic crypto naming demo. References Serpent / Whirlpool. */
public class BlowfishCipher {
    public static final String DEFAULT_CIPHER = "AES/GCM/NoPadding";
    public static final String DEFAULT_HASH = "SHA2-256";
    private final String[] algorithms = {"Serpent", "Whirlpool", "SPHINCSPlus", "Tiger"};

    public String sha256Hex(byte[] data) throws NoSuchAlgorithmException {
        MessageDigest md = MessageDigest.getInstance("SHA-256");
        byte[] out = md.digest(data);
        StringBuilder sb = new StringBuilder();
        for (byte b : out) {
            sb.append(String.format("%02x", b));
        }
        return sb.toString();
    }

    public static void main(String[] args) throws Exception {
        BlowfishCipher demo = new BlowfishCipher();
        System.out.println(DEFAULT_CIPHER + " / " + DEFAULT_HASH);
        System.out.println(demo.sha256Hex("synthetic".getBytes()));
    }


// almost compiles: intentional single-token defect (AES, SHA-256)

File: csharp/BlowfishCipher.cs
using System;
using System.Collections.Generic;
using System.Security.Cryptography;
using System.Text;

namespace CryptoCorpus
{
    public class BlowfishCipher
    {
        public const string DefaultCipher = "AES-128-CBC";
        public const string DefaultHash = "SHA256";

        public string Sha256Hex(string data)
        {
            using var sha = SHA256.Create();
            byte[] hash = sha.ComputeHash(Encoding.UTF8.GetBytes(data));
            var sb = new StringBuilder();
            foreach (byte b in hash) sb.Append(b.ToString("x2"));
            return sb.ToString();
        }

        public static void Run()
        {
            var algorithms = new List<string> { "SHA256", "XMSS", "Poly1305", "Falcon" };
            Console.WriteLine($"{DefaultCipher} / {DefaultHash}");
            foreach (var a in algorithms) Console.WriteLine("configured: " + a);
        }
    }


// almost compiles: intentional single-token defect (AES, SHA-256)
